server/src/main/java/org/opensearch/index/translog/transfer/TranslogCheckpointTransferSnapshot.java

```diff
@@ -136,9 +144,11 @@ public TranslogCheckpointTransferSnapshot build() throws IOException {
             translogTransferSnapshot.setMinTranslogGeneration(highestGenMinTranslogGeneration);
 
             assert this.primaryTerm == highestGenPrimaryTerm : "inconsistent primary term";
-            assert this.generation == highestGeneration : "inconsistent generation";
+            assert this.generation == highestGeneration : " inconsistent generation ";
+            final long finalHighestGeneration = highestGeneration;
```

Comment:
When is `finalHighestGeneration` different from `generation`, just curious?

Reply:
It is not different from `highestGeneration`.  Had to create a final variable to use in lambda expression of assert. 

```
Variable used in lambda expression should be final or effectively final
```